src/main/java/com/blibli/future/service/ReportServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+package com.blibli.future.service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.blibli.future.enums.AbsencePermit;
+import com.blibli.future.enums.LateCondition;
+import com.blibli.future.model.EmployeeSubstitutionLeaveRight;
+import com.blibli.future.repository.AttendanceRepository;
+import com.blibli.future.repository.EmployeeAbsencePermitRepository;
+import com.blibli.future.repository.EmployeeRepository;
+import com.blibli.future.repository.EmployeeSubstitutionLeaveRightRepository;
+import com.blibli.future.repository.EmployeeYearlyLeaveRepository;
+import com.blibli.future.service.api.ReportService;
+import com.blibli.future.vo.ReportVo;
+import com.blibli.future.vo.SingleReportVo;
+
+@Service
+public class ReportServiceImpl implements ReportService{
+	EmployeeRepository employeeRepository;
+	AttendanceRepository attendanceRepository;
+	EmployeeAbsencePermitRepository employeeAbsencePermitRepository;
+	EmployeeYearlyLeaveRepository employeeYearlyLeaveRepository;
+	EmployeeSubstitutionLeaveRightRepository employeeSubstitutionLeaveRightRepository;
+	
+	@Autowired 
+	public ReportServiceImpl(EmployeeRepository employeeRepository, AttendanceRepository attendanceRepository, EmployeeAbsencePermitRepository employeeAbsencePermitRepository, 
+			EmployeeYearlyLeaveRepository employeeYearlyLeaveRepository, EmployeeSubstitutionLeaveRightRepository employeeSubstitutionLeaveRightRepository){
+		this.employeeRepository = employeeRepository;
+		this.attendanceRepository = attendanceRepository;
+		this.employeeAbsencePermitRepository = employeeAbsencePermitRepository;
+		this.employeeYearlyLeaveRepository = employeeYearlyLeaveRepository;
+		this.employeeSubstitutionLeaveRightRepository = employeeSubstitutionLeaveRightRepository;
+	}
+
+	@Override
+	public List<SingleReportVo> ReportParse(List<Object[]> objects) {
+		if(objects!=null)
+		{
+			List<SingleReportVo> singleReportVos = new ArrayList<>();
+			for(Object[] object : objects){
+				SingleReportVo singleReportVo = new SingleReportVo(object);
+				singleReportVos.add(singleReportVo);
+			}
+			return singleReportVos;
+		}
+		return null;
+	}
+
+	@Override
+	public List<ReportVo> SummaryReport(String dept, LocalDate startDate, LocalDate endDate) {
+		List<ReportVo> reports = employeeRepository.initReport(dept);
+		List<Object[]> daysComingObject = attendanceRepository.countEmployeeAttendance(dept, startDate, endDate);
```

Comment:
I'm sorry for not clear enough, I mean something more like `getFullReport()` or `generateFullReport`whichever explains what the method does.